services/events/service.go

```diff
@@ -0,0 +1,53 @@
+package events
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	api "github.com/containerd/containerd/api/services/events"
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/plugin"
+	"google.golang.org/grpc"
+)
+
+func init() {
+	plugin.Register("events-grpc", &plugin.Registration{
+		Type: plugin.GRPCPlugin,
+		Init: func(ic *plugin.InitContext) (interface{}, error) {
+			return NewService(ic.Emitter), nil
+		},
+	})
+}
+
+type Service struct {
+	emitter  *events.Emitter
+	timeouts map[string]*time.Timer
+}
+
+func NewService(e *events.Emitter) api.EventsServer {
+	return &Service{emitter: e}
+}
+
+func (s *Service) Register(server *grpc.Server) error {
+	api.RegisterEventsServer(server, s)
+	return nil
+}
+
+func (s *Service) EventStream(req *api.EventStreamRequest, srv api.Events_EventStreamServer) error {
```

Comment:
I've updated to make the client stream to only receive events from the client namespace.